### PR TITLE
powerd: Creating recurring powerd callback

### PR DESCRIPTION
**Pull request by g2pop**
This CL moves the battery information retrieval into a new function,
called "fu_plugin_powerd_callback()", which returns the information in a
struct.

This information is then used in fu_plugin_update_prepare() for battery
checks. There is also a g_timeout_add() function being created in
fu_plugin_startup() in order to enable the plugin to repeatedly send a method
call to powerd.

fu_plugin_update_prepare is removed as its functionality is now handled
by src/fu-engine.c

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/powerd/fu-plugin-powerd.c [hunk 1]
```diff
@@ -58,71 +95,9 @@ fu_plugin_startup (FuPlugin *plugin, GError **error)
...
-		       "(uud)",
-		       &power_type,
-		       &current_state,
-		       &current_level);
-
-	/* checking if percentage is invalid */
-	if (current_level < 1 || current_level > 100)
-		current_level = FU_BATTERY_VALUE_INVALID;
+	/* start a timer to repeatedly run fu_plugin_powerd_callback at a set interval */
+	g_timeout_add(5000, fu_plugin_powerd_callback, plugin);
```
Comment (hughsie) [inline plugins/powerd/fu-plugin-powerd.c:100]: You can save a little power using `g_timeout_add_seconds(5,` -- we don't really care if it's 4 seconds or 6 seconds really.

Comment (hughsie) [inline plugins/powerd/fu-plugin-powerd.c:100]: Also, `fu_plugin_powerd_refresh_cb` might be a more usual name for a callback

File: plugins/powerd/fu-plugin-powerd.c [hunk 2]
```diff
@@ -28,10 +28,47 @@ fu_plugin_destroy (FuPlugin *plugin)
...
+{
+	FuPlugin *plugin = (FuPlugin *)pointer;
+	FuPluginData *data = fu_plugin_get_data(plugin);
+	FuContext *ctx = fu_plugin_get_context(plugin);
+	guint32 power_type;
+	guint32 current_state;
+	gdouble current_level;
+	g_autoptr(GVariant) powerd_response = NULL;
+
+	/*retrieving battery info from "GetBatteryState" method call to powerd */
```
Comment (hughsie) [inline plugins/powerd/fu-plugin-powerd.c:42]: space after the `*` -- @djcampello is that possible to encode into the clang style checks?

File: plugins/powerd/fu-plugin-powerd.c [hunk 3]
```diff
@@ -28,10 +28,47 @@ fu_plugin_destroy (FuPlugin *plugin)
 		g_object_unref (data->proxy);
 }
 
+static gboolean
+fu_plugin_powerd_callback(gpointer pointer)
```
Comment (hughsie) [inline plugins/powerd/fu-plugin-powerd.c:32]: just use `FuPlugin *plugin` as the argument.

File: plugins/powerd/fu-plugin-powerd.c (hunk 1 above)
Comment (hughsie) [reply plugins/powerd/fu-plugin-powerd.c:100]: Also, use G_CALLBACK(fu_plugin_powerd_refresh_cb) and then you don't have to do the gpointer cast above.

File: plugins/powerd/fu-plugin-powerd.c [hunk 4]
```diff
@@ -58,71 +94,9 @@ fu_plugin_startup (FuPlugin *plugin, GError **error)
...
-		       "(uud)",
-		       &power_type,
-		       &current_state,
-		       &current_level);
-
-	/* checking if percentage is invalid */
-	if (current_level < 1 || current_level > 100)
-		current_level = FU_BATTERY_VALUE_INVALID;
+	/* start a timer to repeatedly run fu_plugin_powerd_refresh_cb at a set interval */
+	g_timeout_add_seconds(5, G_CALLBACK(fu_plugin_powerd_refresh_cb), plugin);
```
Comment (djcampello) [inline plugins/powerd/fu-plugin-powerd.c:99]: I think here you want G_SOURCE_FUNC() instead of G_CALLBACK()

File: plugins/powerd/fu-plugin-powerd.c [hunk 5]
```diff
@@ -28,10 +28,46 @@ fu_plugin_destroy (FuPlugin *plugin)
...
+	fu_context_set_battery_level(ctx, current_level);
+
+	return G_SOURCE_CONTINUE;
+}
+
 gboolean
 fu_plugin_startup (FuPlugin *plugin, GError **error)
 {
-	FuPluginData *data = fu_plugin_get_data (plugin);
+	FuPluginData *data = fu_plugin_get_data(plugin);
```
Comment (djcampello) [inline plugins/powerd/fu-plugin-powerd.c:70]: Unrelated change

File: plugins/powerd/fu-plugin-powerd.c (hunk 4 above)
Comment (hughsie) [reply plugins/powerd/fu-plugin-powerd.c:99]: Oops, yes, what he said -- sorry!

**Review by hughsie — DISMISSED** [review]

**Comment by djcampello** [comment]
I added a CL to pre format the whole file since @g2pop was having troubles getting the CI happy about formatting

**Review by hughsie — APPROVED** [review]

**Comment by superm1** [comment]
> I added a CL to pre format the whole file since @g2pop was having troubles getting the CI happy about formatting

Hopefully that wasn't caused by my changes in https://github.com/fwupd/fwupd/commit/8b679f7320d7611eb0928be1a594f90a8f112c3b
That was supposed to make things easier...
